share/completions/mkdir.fish

```diff
@@ -0,0 +1,9 @@
+complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
+complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
+complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
+complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
```

Comment:
>But if you can only manage to provide completions relevant to a Linux modinfo, it seems reasonable to check uname. 

It seems like the linux one has `--version` and the solaris one doesn't - long options started as a GNUism, so other OSs don't have them, and if they do they're most likely compatible with the GNU stuff.

We should really try not to call uname - as I said, hurd will have GNU tools but won't identify as linux.

>I think perhaps further you should also test, somehow, for SELinux on Linux (or, at least, a SELinux-enabled mkdir), and if that doesn't exist, don't show the --context=CTX (-Z CTX) options. 

It seems like the canonical way to check if SELinux is enabled is `sestatus`. Since I think `mkdir` can also add contexts even if selinux is installed but not enabled, we should get away with just checking `command -s sestatus`.